infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -230,8 +230,7 @@ public boolean apply(LogReplicationEntryMsg msg) throws ReplicationWriterExcepti
             lastMsgTs = srcGlobalSnapshot;
         }
 
-        // If the entry is the expected entry, process it and process the messages in the queue.
-        if (msg.getMetadata().getPreviousTimestamp() == lastMsgTs) {
+        if (msg.getMetadata().getPreviousTimestamp() <= lastMsgTs && msg.getMetadata().getTimestamp() > lastMsgTs) {
```

Comment:
nit: `if (msg.getMetadata().getPreviousTimestamp() <= lastMsgTs && lastMsgTs < msg.getMetadata().getTimestamp())` for slightly better readability.